Plugin~/Src/MeshSync/msServer.cpp

```diff
@@ -608,6 +619,56 @@ void Server::recvPoll(HTTPServerRequest& request, HTTPServerResponse& response)
     }
 }
 
+void Server::recvServerInitiatedRequest(HTTPServerRequest& request, HTTPServerResponse& response)
+{
+    auto mes = deserializeMessage<ServerInitiatedMessage>(request, response);
+
+    queueMessage(mes);
+
+    // wait for data arrive
+    for (int i = 0; ; ++i) {
+        if (mes->ready)
+            break;
+        if (mes->cancelled)
+            return;
+
+        std::this_thread::sleep_for(std::chrono::milliseconds(10));
+    }
+
+    // serve data
+    response.set("Cache-Control", "no-store, must-revalidate");
+    
+    auto reqResponse = ServerInitiatedMessageResponse();
+
+    for (auto prop : m_pending_properties)
+    {
+        auto p = PropertyInfo(*prop);
+        reqResponse.properties.push_back(p);
+    }
+    m_pending_properties.clear();
+
+    auto converters = Scene::getConverters(m_settings.import_settings, m_current_properties_request->scene_settings, true);
+
+    for (auto entity : m_pending_entities) {
+        for (auto& cv : converters) {
+            cv->convert(*entity);
+        }
+
+        reqResponse.entities.push_back(entity);
+    }
+
+    m_pending_entities.clear();
+    
+    if (m_syncRequested) {
+        reqResponse.message = REQUEST_SYNC;
```

Comment:
Clarifying, `REQUEST_SYNC` requests the client to send data back to Unity after it receives `ServerInitiatedMessageResponse` ?

Reply:
Yes